### PR TITLE
Include ijwhost in DesignerRuntimeImplementationProjectOutputGroupOutput

### PR DESCRIPTION
**Pull request by elinor-fung**
`ijwhost` is required for C++/CLI assemblies. If a project that has a designer (like WinForms) references a C++/CLI assembly, it needs to be told about `ijwhost`.

It looks like there is no automated testing support for C++/CLI projects right now due to https://github.com/dotnet/sdk/issues/11008. I just manually ran `msbuild /t:DesignerRuntimeImplementationProjectOutputGroup` on a C++/CLI project and checked that ijwhost was in the target outputs.

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets [hunk 1]
```diff
@@ -139,6 +139,13 @@ Copyright (c) .NET Foundation. All rights reserved.
         Include="@(_DesignerShadowCopy->'%(FullPath)')"
         TargetPath="%(_DesignerShadowCopy.DestinationSubDirectory)%(_DesignerShadowCopy.Filename)%(_DesignerShadowCopy.Extension)"
         />
+
+      <!-- Include ijwhost.dll for projects that use it -->
+      <DesignerRuntimeImplementationProjectOutputGroupOutput
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets:144]: It seems like this target would need to depend on `ResolveFrameworkReferences` in order to correctly have the `IjwHostSourcePath` set.

**Review by dsplaisted — APPROVED** [review]
Would it be possible to validate the end-to-end scenario where this will fix the designer loading?

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets (hunk 1 above)
Comment (elinor-fung) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets:144]: From locally building, it does have that via a dependency chain of `DesignerRuntimeImplementationProjectOutputGroup` > `AssignTargetPaths` > `_ComputeNETCoreBuildOutputFiles` > `_GetIjwHostPaths` > `ResolveFrameworkReferences`

**Comment by elinor-fung** [comment]
> Would it be possible to validate the end-to-end scenario where this will fix the designer loading?

I asked @Shyam-Gupta over e-mail if he could validate.

**Review by dsplaisted — APPROVED** [review]
Looks good.  Note that the main branch is for the .NET 8 SDK, you should probably retarget this to `release/7.0.2xx` if you want this to go in the next feature release of the SDK.

**Comment by elinor-fung** [comment]
Thanks - retargeted. I don't know much about the release/branching in sdk - is `release/7.0.2xx` open or is there some sort of approval process? Does it flow back to `main` or would I need to port it?

**Comment by dsplaisted** [comment]
`release/7.0.2xx` is open and changes will automatically flow into main (eventually).

**Pull request merged**